test/graphql/samples_query_test.rb

```diff
@@ -34,4 +62,33 @@ def setup
     assert_not_empty data, 'samples type should work'
     assert_not_empty data['nodes']
   end
+
+  test 'sample and metadata fields query should work' do
+    result = IridaSchema.execute(SAMPLE_AND_METADATA_QUERY, context: { current_user: @user },
+                                                            variables: { sample_id: @sample32.to_global_id.to_s })
+
+    assert_nil result['errors'], 'should work and have no errors.'
+
+    data = result['data']['sample']
+
+    assert_not_empty data, 'sample type should work'
+    assert_equal @sample32.metadata, data['metadata']
+    assert data['metadata'].key?('metadatafield1')
+    assert data['metadata'].key?('metadatafield2')
+  end
+
+  test 'sample and metadata fields query should be able to limit to a specific set of keys' do
+    result = IridaSchema.execute(SAMPLE_AND_LIMITED_METADATA_QUERY,
+                                 context: { current_user: @user },
+                                 variables: { sample_id: @sample32.to_global_id.to_s, keys: ['metadatafield1'] })
+
+    assert_nil result['errors'], 'should work and have no errors.'
+
+    data = result['data']['sample']
+
+    assert_not_empty data, 'sample type should work'
+    assert_not_equal @sample32.metadata, data['metadata']
+    assert data['metadata'].key?('metadatafield1')
+    assert_not data['metadata'].key?('metadatafield2')
+  end
```

Comment:
Can I just get you to add one more test here for the case of a user querying for a metadata key that isn't in the sample metadata

Reply:
Thanks!